tests/generation/test_utils.py

```diff
@@ -153,12 +157,16 @@ def _get_logits_processor_and_kwargs(
                 if forced_eos_token_id is not None
                 else []
             )
-            + [
-                NoBadWordsLogitsProcessor(process_kwargs["bad_words_ids"], eos_token_id),
-                NoRepeatNGramLogitsProcessor(process_kwargs["no_repeat_ngram_size"]),
-                RepetitionPenaltyLogitsProcessor(process_kwargs["repetition_penalty"]),
-            ]
+            + [NoBadWordsLogitsProcessor(process_kwargs["bad_words_ids"], eos_token_id)]
+            + (
+                NoRepeatNGramLogitsProcessor(process_kwargs["no_repeat_ngram_size"])
+                if forced_bos_token_id is None and forced_eos_token_id is None
+                else []
+            )
+            + [RepetitionPenaltyLogitsProcessor(process_kwargs["repetition_penalty"])]
+            + [InfNanRemoveLogitsProcessor()]  # prevent flaky generation test failures
```

Comment:
This one was being added appended before the `sample` call, and `generate` had `remove_invalid_values=True` (which triggers this logits processor).

Since `generate` appends logits processors passed with the `logits_processor` kwarg at the end of the list, this meant that prior to this change, `generate` was calling `InfNanRemoveLogitsProcessor` as the first logits processor, and `sample` as the last one. 

Order of operations matters in the logits processors! :)